addons/mail/static/src/components/thread_view/thread_view_tests.js

```diff
@@ -649,6 +653,146 @@ QUnit.test('basic rendering of canceled notification', async function (assert) {
+    );
+    assert.containsNone(
+        document.body,
+        '.o_Message',
+        "there should be no message left displayed"
+    );
+});
+
+QUnit.test('delete all attachments of a message with some text content should still keep it displayed', async function (assert) {
+    assert.expect(4);
+
+    this.data['ir.attachment'].records.push({
+        id: 143,
+        mimetype: 'text/plain',
+        name: "Blah.txt",
+    });
+    this.data['mail.channel'].records.push({ id: 11 });
+    this.data['mail.message'].records.push(
+        {
+            attachment_ids: [143],
+            body: "hey man !",
+            channel_ids: [11],
+            id: 100,
+        }, {
+            id: 101,
+            channel_ids: [11],
+            tracking_value_ids: [6]
+        },
+    );
+    this.data['mail.tracking.value'].records.push({
+        changed_field: "Name",
+        field_type: "char",
+        id: 6,
+        new_value: "New name",
+        old_value: "Old name",
+    });
+    await this.start();
+    const threadViewer = this.env.models['mail.thread_viewer'].create({
+        hasThreadView: true,
+        thread: [['insert', { id: 11, model: 'mail.channel' }]],
+    });
+    // wait for messages of the thread to be loaded
+    await this.afterEvent({
+        eventName: 'o-component-message-list-thread-cache-changed',
+        func: () => {
+            this.createThreadViewComponent(threadViewer.threadView);
+        },
+        message: "thread become loaded with messages",
+        predicate: ({ threadViewer }) => {
+            return threadViewer.thread.model === 'mail.channel' && threadViewer.thread.id === 11;
+        },
+    });
+    assert.containsN(
+        document.body,
+        '.o_Message',
+        2,
+        "there should be 2 messages displayed"
+    );
+
+    await afterNextRender(() => {
+        document.querySelector(`.o_Attachment[data-attachment-local-id="${
+            this.env.models['mail.attachment'].findFromIdentifyingData({ id: 143 }).localId
+        }"] .o_Attachment_asideItemUnlink`).click();
+    });
+    await afterNextRender(() =>
+        document.querySelector('.o_AttachmentDeleteConfirmDialog_confirmButton').click()
+    );
+    assert.containsN(
+        document.body,
+        '.o_Message',
+        2,
+        "there should still be 2 messages left displayed"
+    );
+    assert.containsOnce(
+        document.body,
+        `.o_Message[data-message-local-id="${
+            this.env.models['mail.message'].findFromIdentifyingData({ id: 100 }).localId
+        }"]`,
+        "the message with text content should be left displayed"
+    );
+    assert.containsOnce(
+        document.body,
+        `.o_Message[data-message-local-id="${
+            this.env.models['mail.message'].findFromIdentifyingData({ id: 101 }).localId
+        }"]`,
+        "the message without text content but with tracking values should be left displayed"
+    );
```

Comment:
Same remark as other test: single message, _"[...] initially"_ and _"[...] after deleting all (1) attachments of message"_.